qiskit_nature/problems/sampling/protein_folding/interactions/random_interaction.py

```diff
@@ -21,19 +21,18 @@
 class RandomInteraction(Interaction):
     """A class defining a random interaction between beads of a peptide."""
 
-    def calculate_energy_matrix(
-        self, chain_len: int, residue_sequence: List[str] = None
-    ) -> np.ndarray:
+    # TODO only needs length...
```

Comment:
Do you want to address this as part of this PR or in the future?

Reply:
Ideally we could solve it immediately, but I am not sure what is the best approach here.

Reply:
I have exactly the same question as @mrossinek  of the type change. In the previous version one would define the residues as `List[str]`. How can one define it in a single `Str` ?

Reply:
Ok, for a moment I got confused and thought that a residue provided might potentially be coming from a main chain or a side chain. It might be the case in the future when extended to longer side chain but currently we only rely on the main chain. I will revert this change to avoid confusion for now.

Reply:
@mrossinek The problem is that the random interaction does not use residues. It only needs the length of a residue. Ideally, it would only accept the length to avoid confusion that changing a residue would change the result somehow... 
But then it does not conform to the abstract method definition in the Interaction class...

But maybe it is not too bad...

Reply:
Yes I see. I think this is a very special case that the RandomInteraction does not make use of all the information given to it (i.e. it only cares about the length of the list and not its actual contents). This can happen and I don't think it's a problem.

And since we have a clear abstract definition, there is nothing we can do about it :upside_down_face: Simply keep the method definition, get the length from the list and ignore the rest. You can add a comment that this method does not care about the contents but other than that I would not change anything